### PR TITLE
doc: update documentation for 1.4.1

### PR DESCRIPTION
**Pull request by greg-fer**
Updated documentation for 1.4.1 release.
Bumped relese version number.
Added files for release notes.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

File: doc/versions.json [hunk 1]
```diff
@@ -19,6 +20,12 @@
       "zephyr": "2.4.0",
       "mcuboot": "1.6.99"
     },
+    "1.4.1": {
+      "ncs": "1.4.1",
+      "nrfxlib": "1.4.0",
+      "zephyr": "2.4.0-ncs2",
```
Comment (ru-fu) [inline doc/versions.json:26]: This is a tag, but not a documentation version.
It should either be 2.4.99 (if there are any changes) or 2.4.0.

@carlescufi , you discussed that with @joerchan for the v1.3.2 release and said the Zephyr version should stay as is, even if there's a new tag?

File: doc/nrf/links.txt [hunk 2]
```diff
@@ -52,6 +54,8 @@
 
 .. _`known issues for nRF Connect SDK v1.4.0`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/known_issues.html?v=v1-4-0
 
+.. _`known issues for nRF Connect SDK v1.4.1`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/known_issues.html?v=v1-4-1
```
Comment (ru-fu) [inline doc/nrf/links.txt:57]: Make sure to add that version on master, otherwise the filtering won't work.

File: doc/nrf/links.txt [hunk 3]
```diff
@@ -34,6 +34,8 @@
 
 .. ### Links to Nordic documentation
 
+.. _`nRF Connect SDK v1.4.0 documentation`: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/1.4.0/nrf/index.html
```
Comment (ru-fu) [inline doc/nrf/links.txt:37]: Why is this one needed?

**Review by ru-fu — COMMENTED** [review]
The changes to versions.json are mostly needed on master (since the file in this release isn't used). It makes sense to update the file here as well, but this won't add 1.4.1 to the navigation.

File: doc/nrf/links.txt (hunk 2 above)
Comment (greg-fer) [reply doc/nrf/links.txt:57]: Yes, planned in this PR: https://github.com/nrfconnect/sdk-nrf/pull/3382

File: doc/nrf/links.txt (hunk 3 above)
Comment (greg-fer) [reply doc/nrf/links.txt:37]: Will remove.

File: doc/versions.json (hunk 1 above)
Comment (greg-fer) [reply doc/versions.json:26]: Talked about this with @rlubos , and there are no changes that we're including from upstream except some cherry-picks. Will revert to 2.4.0.

File: doc/nrf/releases/release-notes-1.4.1.rst [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+* Qualified Zephyr Host to be conformant with Bluetooth Core Specification v5.2.
+  These changes were cherry-picked in the |NCS| v1.4.1 Bluetooth Host qualification update, which contains the following changes:
+
+  * Added qualification for Bluetooth Core Specification v5.2.
+    Qualified new features are the following:
+
+    * Advertising extension
+    * Legacy OOB pairing
+
+* Fixed an issue where a pairing fail could lead to the GATT procedure failure.
```
Comment (joerchan) [inline doc/nrf/releases/release-notes-1.4.1.rst:104]:
```suggestion
* Fixed an issue where a pairing fail could lead to GATT procedure failure.
```

Comment (joerchan) [reply doc/nrf/releases/release-notes-1.4.1.rst:104]: or maybe "a"

**Review by kromancer — APPROVED** [review]

**Review by ru-fu — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by nordic-hani — APPROVED** [review]

File: doc/nrf/releases/release-notes-1.4.1.rst [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+The |NCS| is where you begin building low-power wireless applications with Nordic Semiconductor nRF52, nRF53, and nRF91 Series devices.
+nRF53 Series devices (which are pre-production) and Zigbee and Bluetooth mesh protocols are supported for development in v1.4.1 for prototyping and evaluation.
+Support for production and deployment in end products is coming soon.
+
+
+Highlights
+**********
+
+* Qualified Zephyr Host to be conformant with Bluetooth Core Specification v5.2
```
Comment (jdstapes) [inline doc/nrf/releases/release-notes-1.4.1.rst:21]: "Zephyr Host" should be "BLE Host Subsystem" in all cases

Comment (greg-fer) [reply doc/nrf/releases/release-notes-1.4.1.rst]: Updated using "Bluetooth Host" instead of "BLE Host" after checking with @joerchan .

**Review by jdstapes — APPROVED** [review]

**Comment by joerchan** [comment]
@greg-fer documentation build failed. can you have a look?

**Comment by carlescufi** [comment]
@greg-fer still failing, something missing here?

File: doc/nrf/known_issues.rst [hunk 6]
```diff
@@ -212,6 +286,22 @@ Subsystems
...
 
+.. rst-class:: v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0 v1-2-1 v1-2-0 v1-1-0 v1-0-0
+
+NCSDK-6845: Pairing failure with simultaneous pairing on multiple connections
+  When using LE Secure Connections pairing, the pairing fails with simultaneous pairing on multiple connections.
+  The failure reason is unspecified.
+
+  **Workaround:** Retry the pairing on the connections that failed one by one after the pairing procedure has finished.
+
+.. rst-class:: v1-4-1 v1-4-0 v1-3-2 v1-3-1 v1-3-0
```
Comment (joerchan) [inline doc/nrf/known_issues.rst:297]: This issue has been fixed for 1.4.1

See bottom:
`* Fixed an issue where a pairing fail could lead to a GATT procedure failure.`

**Review by joerchan — APPROVED** [review]

**Pull request merged**